frontend/src/components/LocationForm.jsx

```diff
@@ -0,0 +1,56 @@
+// src/components/LocationForm.jsx
+import React, { useState } from 'react';
+import styled from 'styled-components';
+
+const Form = styled.form`
+  display: flex;
+  gap: 10px;
+  margin-bottom: 20px;
+`;
+
+const Input = styled.input`
+  padding: 10px;
+  border: 1px solid #ccc;
+  border-radius: 5px;
+  flex: 1;
+`;
+
+const Button = styled.button`
+  padding: 10px 20px;
+  background-color: #007bff;
+  color: #fff;
+  border: none;
+  border-radius: 5px;
+  cursor: pointer;
+
+  &:hover {
+    background-color: #0056b3;
+  }
+`;
+
+const LocationForm = () => {
```

Comment:
Consider optimizing the API call frequency if the user enters multiple characters quickly in the search input.